### PR TITLE
#7 - Apps have ACLs

### LINKED ISSUE #8: Apps should permit ACL configuration
ACL should be considered when building the App's package content, and also returned by an appropriate data source with the App's attributes so that the ACL can be managed using `splunk_generic_acl`.

### PR DESCRIPTION
**Pull request by micahkemp-splunk**
Closes #8 

This PR adds a new abstraction type `ACL`, which has been added to the `App` type.

The `ACL` type will likely be added to other types later on, but the immediate need is to be able to manage permissions of apps, so I've only added it there for now.

These changes allow a templated app to have a `default.meta` file, and allow querying an `App`'s attributes, to pass on to the Splunk Terraform provider as appropriate.

### CODE REVIEW

File: internal/splunkconfig/config/sharing.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	return nil
+}
+
+// metaValue returns a string for use in a .meta file's "export" configuration.
+func (sharing Sharing) metaValue() string {
+	if sharing == SHAREGLOBAL {
+		return "system"
+	}
+
+	return ""
```
Comment (guohuang) [inline internal/splunkconfig/config/sharing.go:48]: This method only returns "system" or empty string? what about other values?

File: internal/splunkconfig/config/sharing_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+			false,
+		},
+		{
+			Sharing("global"),
+			false,
+		},
+	}
+
+	tests.test(t)
+}
```
Comment (guohuang) [inline internal/splunkconfig/config/sharing_test.go:45]: how about adding a test for the `metaValue ` method?

File: internal/splunkconfig/config/sharing.go (hunk 1 above)
Comment (micahkemp-splunk) [reply internal/splunkconfig/config/sharing.go:48]: Good question.  There are no other values:

https://docs.splunk.com/Documentation/Splunk/8.2.2/Admin/Defaultmetaconf

```
* By default, objects are only visible within the app in which they were created.
  To make an object available to all apps, set the object's 'export' setting to
  "system".
  * export = system
```

It's either `export = ` (empty or unset) or `export = system`.

**Review by guohuang — APPROVED** [review]

File: internal/provider/appattributes.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import (
+	"context"
+	"terraform-provider-splunkconfig/internal/splunkconfig/config"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	appAttributesIdKey              = "app_id"
```
Comment (mbeacom-splunk) [inline internal/provider/appattributes.go:26]: Any reason for `appAttributesIdKey` and `app_id` here as opposed to `appAttributesAppIdKey`?

File: internal/provider/appattributes.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+func resourceAppAttributesRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	suite := meta.(config.Suite)
+
+	appId := d.Get(appAttributesIdKey).(string)
+
+	d.SetId(appId)
+
+	apps, err := suite.ExtrapolatedApps()
+	if err != nil {
```
Comment (mbeacom-splunk) [inline internal/provider/appattributes.go:107]: Still one of my favorite patterns of Go.

**Review by mbeacom-splunk — APPROVED** [review]

File: internal/provider/appattributes.go (hunk 3 above)
Comment (micahkemp-splunk) [reply internal/provider/appattributes.go:26]: Only to be consistent with my previously poorly named constants for other resources.  Sounds like I feel prey to broken windows.

Comment (micahkemp-splunk) [reply internal/provider/appattributes.go]: I've gone and repaired those broken windows.

**Pull request merged**